mapie/conformity_scores/sets/naive.py

```diff
@@ -0,0 +1,506 @@
+        NDArray of shape (n_samples, n_alpha)
+            Vs parameters.
+        """
+        # compute V parameter from Romano+(2020)
+        vs = (
+            (y_proba_last_cumsumed - threshold.reshape(1, -1)) /
+            y_pred_proba_last[:, 0, :]
+        )
+        return vs
+
+    def _add_random_tie_breaking(
+        self,
+        prediction_sets: NDArray,
+        y_pred_index_last: NDArray,
+        y_pred_proba_cumsum: NDArray,
+        y_pred_proba_last: NDArray,
+        threshold: NDArray,
+        random_state: Optional[Union[int, np.random.RandomState]] = None,
+        **kwargs
+    ) -> NDArray:
+        """
+        Randomly remove last label from prediction set based on the
+        comparison between a random number and the difference between
+        cumulated score of the last included label and the quantile.
+
+        Parameters
+        ----------
+        prediction_sets: NDArray of shape
+            (n_samples, n_classes, n_threshold)
+            Prediction set for each observation and each alpha.
+
+        y_pred_index_last: NDArray of shape (n_samples, threshold)
+            Index of the last included label.
+
+        y_pred_proba_cumsum: NDArray of shape (n_samples, n_classes)
+            Cumsumed probability of the model in the original order.
+
+        y_pred_proba_last: NDArray of shape (n_samples, 1, threshold)
+            Last included probability.
+
+        threshold: NDArray of shape (n_alpha,) or shape (n_samples_train,)
+            Threshold to compare with y_proba_last_cumsum, can be either:
+
+            - the quantiles associated with alpha values when
+                ``cv`` == "prefit", ``cv`` == "split"
+                or ``agg_scores`` is "mean"
+
+            - the conformity score from training samples otherwise (i.e., when
+            ``cv`` is CV splitter and ``agg_scores`` is "crossval")
+
+        method: str
+            Method that determines how to remove last label in the prediction
+            set.
+
+            - if "cumulated_score" or "aps", compute V parameter
+                from Romano+(2020)
+
+            - else compute V parameter from Angelopoulos+(2020)
+
+        lambda_star: Optional[Union[NDArray, float]] of shape (n_alpha):
+            Optimal value of the regulizer lambda.
+
+        k_star: Optional[NDArray] of shape (n_alpha):
+            Optimal value of the regulizer k.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes, n_alpha)
+            Updated version of prediction_sets with randomly removed labels.
+        """
+        # get cumsumed probabilities up to last retained label
+        y_proba_last_cumsumed = np.squeeze(
+            np.take_along_axis(
+                y_pred_proba_cumsum,
+                y_pred_index_last,
+                axis=1
+            ), axis=1
+        )
+
+        # get the V parameter from Romano+(2020) or Angelopoulos+(2020)
+        vs = self._compute_vs_parameter(
```

Comment:
there is no vs need in the naive method

Reply:
vs is needed when `include_last_label == "randomized"`, do we have a condition that prohibits `method='naive'` and `include_last_label='randomized'`?

Reply:
yes we should, acutually naive method should always have include_last_label=True

Reply:
This condition is not currently verified, is that correct? Should this therefore be the subject of another PR?

Reply:
I think we can add it in this PR, it will make the code easier to read

Reply:
@LacombeLouis do you agree ?

Reply:
it's not linked

Reply:
Done 👍 (move this method in APS class)